File: src/main/java/july_2025/MoveZeroes.java
package july_2025;

public class MoveZeroes {

    public static void main(String[] args) {
//        int[] nums = {0,1,0,3,12};
        int[] nums = {1};

        MoveZeroes moveZeroes = new MoveZeroes();
        moveZeroes.moveZeroes(nums);
        System.out.println();

    }

    // O(n) time | O(1) space
    public void moveZeroes(int[] nums) {
        int posIdx = 0;
        for (int num : nums) {
            if (num != 0) {
                nums[posIdx] = num;
                posIdx++;
            }
        }
        for (int i = posIdx; i < nums.length; i++) {
            nums[i] = 0;
        }
    }


}

File: python/second/move_zeroes.py
class Solution(object):
    # O(n) time | O(1) space
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        posIdx = 0

        for num in nums:
            if num != 0:
                nums[posIdx] = num
                posIdx += 1
        for i in range(posIdx, len(nums)):
            nums[i] = 0


solution = Solution()
nums = [0,1,0,3,12]
solution.moveZeroes(nums)

print(nums)
